Documentation/guides/AndroidAssetPacks.md

```diff
@@ -0,0 +1,176 @@
+build system which pack to place this asset in. Since auto import of items
+is common now we need a way for a user to add this additional attribute
+to auto included items. Fortunately we are able to use the following.
+
+```xml
+<ItemGroup>
+   <AndroidAsset Update="Asset/movie.mp4" AssetPack="assets1" />
+   <AndroidAsset Update="Asset/movie2.mp4" AssetPack="assets1" />
+   <AndroidAsset Update="Asset/movie3.mp4" AssetPack="assets2" />
+</ItemGroup>
+```
+
+This code uses the `Update` attribute to tell MSBuild that we are going
+to update a specific item. Note in the sample we do NOT need to include
+an `Update` for the `data.xml`, since this is auto imported it will still
+end up in the main feature in the aab.
+
+Additional attributes can be used to control what type of asset pack is
+produced. The only extra one supported at this time is `DeliveryType`,
+this can have a value of `InstallTime`, `FastFollow` or `OnDemand`.
+Additional attributes do not need to be included on ALL items. Any one
+will do, only the `AssetPack` attribute will be needed.
+See Google's [documentation](https://developer.android.com/guide/playcore/asset-delivery#asset-updates) for details on what each item does.
+
+```xml
+<ItemGroup>
+   <AndroidAsset Update="Asset/movie.mp4" AssetPack="assets1" DeliveryType="InstallTime" />
+   <AndroidAsset Update="Asset/movie2.mp4" AssetPack="assets1" />
+   <AndroidAsset Update="Asset/movie3.mp4" AssetPack="assets2" />
+</ItemGroup>
+```
+
+If the `AssetPack` attribute is not present, the default behavior will
+be to include the asset in the main application package.
+
+If however you have a large number of assets it might be more efficient to make use of the `base` asset pack setting. In this scenario you update ALL assets to be in a single asset pack then use the `AssetPack="base"` metadata to declare which specific assets end up in the base aab file. With this you can use wildcards to move most assets into the asset pack.
+
+```xml
+<ItemGroup>
+   <AndroidAsset Update="Assets/*" AssetPack="assets1" />
+   <AndroidAsset Update="Assets/movie.mp4" AssetPack="base" />
+   <AndroidAsset Update="Assets/some.png" AssetPack="base" />
+</ItemGroup>
+```
+
+In this example, `movie.mp4` and `some.png` will end up in the `base` aab file, but ALL the other assets
+will end up in the `assets1` asset pack.
+
+## Implementation Details
+
+There are a few changes we need to make in order to support this feature.
+One of the issues we will hit is the build times when dealing with large assets.
+Current the assets which are to be included in the `aab` are COPIED
+into the `$(IntermediateOutputPath)assets` directory. This folder is
+then passed to `aapt2` for the build process.
+
+The new system adds a new directory `$(IntermediateOutputPath)assetpacks`.
+This directory would contain a subdirectory for each `pack` that the
+user wants to include.
+
+```dotnetcli
+assetpacks/
+    assets1/
+        assets/
+            movie1.mp4
+    feature2/
```

Comment:
What is `feature2`, and why is it named "feature2" when it's a sibling of "assets1"?

Are assets features?  Are features assets?  Is a "feature pack" *really* a "feature asset pack"?